trench/backends/base.py

```diff
@@ -64,7 +66,15 @@ def dispatch_message(self) -> DispatchResponse:
         raise NotImplementedError  # pragma: no cover
 
     def create_code(self) -> str:
-        return self._get_otp().now()
+        # totp
+        if self._mfa_method.is_totp:
+            return self._get_otp().now()
+
+        # hotp
```

Comment:
Perhaps rather than conditional code their should be an HOTP backend that can be subclassed for a handler in the settings.